File: GitToolBox/src/main/java/zielu/gittoolbox/extension/UpdateProjectAction.java
package zielu.gittoolbox.extension;

import com.intellij.openapi.actionSystem.AnAction;
import zielu.gittoolbox.ResBundle;

public interface UpdateProjectAction {
  String getId();

  boolean isDefault();

  default String getName() {
    return ResBundle.message(getId());
  }

  AnAction getAction();
}

File: src/main/kotlin/zielu/gittoolbox/extension/update/UpdateProjectAction.kt
package zielu.gittoolbox.extension.update

import com.intellij.openapi.actionSystem.AnAction
import zielu.gittoolbox.ResBundle

internal interface UpdateProjectAction {
  fun getId(): String

  fun isDefault(): Boolean

  fun getName(): String {
    return ResBundle.message(getId())
  }

  fun getAction(): AnAction
}
